src/Azure.Functions.Cli/Actions/AzureActions/PublishFunctionAppAction.cs

```diff
@@ -276,10 +276,21 @@ public override async Task RunAsync()
                 }
             }
 
-            // Check if azure-functions-worker exists in requirements.txt for Python function app
+            // Special checks for python dependencies
             if (workerRuntime == WorkerRuntime.python)
             {
+                // Check if azure-functions-worker exists in requirements.txt for Python function app
                 await PythonHelpers.WarnIfAzureFunctionsWorkerInRequirementsTxt();
+                // Check if remote LinuxFxVersion exists and is different from local version
+                if (!string.IsNullOrEmpty(functionApp.LinuxFxVersion))
+                {
+                    string localImageVersion = await PythonHelpers.ChoosePythonBuildEnvImage();
+                    if (!string.Equals(localImageVersion, functionApp.LinuxFxVersion))
```

Comment:
The `LinuxFxVersion` cannot be used to compare with the localImageVersion directly. Since localImageVersion and LinuxFxVersion returns different format.
```
functionApp.LinuxFxVersion = "PYTHON|<major>.<minor>"
localImageVersion  = "mcr.microsoft.com/azure-functions/python:2.0.14786-python3.6-buildenv"
```

I would suggest moving this piece of logic into PythonHelper.cs, parsing the <major> and <minor> version from LinuxFxVersion, and compare the <major> <minor> with `GetEnvironmentPythonVersion().Major` `GetEnvironmentPythonVersion().Minor`.

You can find `AssertPythonVersion` as a reference.